examples/playwright/src/theia-notebook-cell.ts

```diff
@@ -0,0 +1,212 @@
+// *****************************************************************************
+import { ElementHandle, FrameLocator, Locator } from '@playwright/test';
+import { TheiaApp } from './theia-app';
+import { TheiaMonacoEditor } from './theia-monaco-editor';
+import { TheiaPageObject } from './theia-page-object';
+
+export type CellStatus = 'success' | 'error' | 'waiting';
+
+/**
+ * Page object for a Theia notebook cell.
+ */
+export class TheiaNotebookCell extends TheiaPageObject {
+
+    protected monacoEditor: TheiaEmbeddedMonacoEditor;
+
+    constructor(protected readonly locator: Locator, app: TheiaApp) {
+        super(app);
+        const editorLocator = locator.locator('div.theia-notebook-cell-editor');
+        this.monacoEditor = new TheiaEmbeddedMonacoEditor(editorLocator, app);
+    }
+
+    /**
+     * @returns The monaco editor page object of the cell.
+     */
+    get editor(): TheiaEmbeddedMonacoEditor {
+        return this.monacoEditor;
+    }
+
+    /**
+     * @returns `true` id the cell is a code cell, `false` otherwise.
+     */
+    async isCodeCell(): Promise<boolean> {
+        const classAttribute = await this.mode();
+        return classAttribute !== 'markdown';
+    }
+
+    /**
+     * @returns The mode of the cell, e.g. 'python', 'markdown', etc.
+     */
+    async mode(): Promise<string> {
+        this.locator.waitFor({ state: 'visible' });
+        const editorElement = await this.editor.locator.elementHandle();
+        if (editorElement === null) {
+            throw new Error('Could not find editor element for the notebook cell.');
+        }
+        const classAttribute = await editorElement.getAttribute('data-mode-id');
+        if (classAttribute === null) {
+            throw new Error('Could not find mode attribute for the notebook cell.');
+        }
+        return classAttribute;
+    }
+
+    /**
+     * @returns The text content of the cell editor.
+     */
+    async editorText(): Promise<string | undefined> {
+        const lines: string[] = [];
+        const linesCount = await this.monacoEditor.numberOfLines();
+        if (linesCount === undefined) {
+            return undefined;
+        }
+        for (let line = 1; line <= linesCount; line++) {
+            const lineText = await this.monacoEditor.textContentOfLineByLineNumber(line);
+            if (lineText === undefined) {
+                break;
+            }
+            lines.push(lineText);
+        }
+        return lines.join('\n');
+    }
+
+    /**
+     * Adds text to the editor of the cell.
+     * @param text  The text to add to the editor.
+     * @param lineNumber  The line number where to add the text. Default is 1.
+     */
+    async addEditorText(text: string, lineNumber: number = 1): Promise<void> {
+        const line = await this.editor.lineByLineNumber(lineNumber);
+        await line?.click();
+        await this.page.keyboard.type(text);
```

Comment:
As above, I think it could make sense to move this behavior into the `TheiaMonacoEditor`.